lib/actions/preload.js

```diff
@@ -178,11 +208,73 @@ Alternatively you can pass the --pin-device-to-release flag to pin only this dev
 		});
 };
 
+/**
+ * @param {import('balena-sdk').BalenaSDK} balenaSdk
+ * @param {string | number} appId
+ * @returns {Promise<import('balena-sdk').Application>}
+ */
+async function getAppWithReleases(balenaSdk, appId) {
+	return balenaSdk.models.application.get(appId, {
+		$expand: applicationExpandOptions,
+	});
+}
+
+async function prepareAndPreload(preloader, balenaSdk, options) {
+	const { ExpectedError } = require('../errors');
+
+	await preloader.prepare();
+	preloader.config = { deviceType: 'intel-nuc' };
+
+	const application = options.appId
+		? await getAppWithReleases(balenaSdk, options.appId)
+		: await selectApplication(preloader.config.deviceType);
```

Comment:
The `getAppWithReleases` call is new: previously, `balena-preload` would do the application fetching.